.circleci/config.yml

```diff
@@ -1,64 +1,136 @@
 version: 2
 
+rspec: &rspec
+  steps:
+    - checkout
+    - run: bundle install
+    - run: rake spec
+
+rubocop: &rubocop
+  # Cache key e.g. "rubocop-v1-rubocop-ruby-2.5
+  steps:
+    - checkout
+    - run: bundle install
+    - restore_cache:
+        keys:
+          - rubocop-v1-{{ .Environment.CIRCLE_JOB }}
+          - rubocop-v1-
+    - run: rake internal_investigation
+    - save_cache:
+        key: rubocop-v1-{{ .Environment.CIRCLE_JOB }}
+        paths:
+          - ~/.cache/rubocop_cache
+
 jobs:
-  ruby-2.1:
+  confirm_config:
     docker:
-      - image: circleci/ruby:2.1
+      - image: circleci/ruby:2.5
     steps:
       - checkout
       - run: bundle install
-      - run: rake
-  ruby-2.2:
+      - run: rake confirm_config
+
+  # Ruby 2.1
+  ruby-2.1-rspec:
+    docker:
+      - image: circleci/ruby:2.1
+    <<: *rspec
+  ruby-2.1-rubocop:
+    docker:
+      - image: circleci/ruby:2.1
+    <<: *rubocop
+
+  # Ruby 2.2
+  ruby-2.2-rspec:
     docker:
       - image: circleci/ruby:2.2
-    steps:
-      - checkout
-      - run: bundle install
-      - run: rake
-  ruby-2.3:
+    <<: *rspec
+  ruby-2.2-rubocop:
+    docker:
+      - image: circleci/ruby:2.2
+    <<: *rubocop
+
+  # Ruby 2.3
+  ruby-2.3-rspec:
     docker:
       - image: circleci/ruby:2.3
-    steps:
-      - checkout
-      - run: bundle install
-      - run: rake
-  ruby-2.4:
+    <<: *rspec
+  ruby-2.3-rubocop:
+    docker:
+      - image: circleci/ruby:2.3
+    <<: *rubocop
+
+  # Ruby 2.4
+  ruby-2.4-rspec:
     docker:
       - image: circleci/ruby:2.4
-    steps:
-      - checkout
-      - run: bundle install
-      - run: rake
-  ruby-2.5:
+    <<: *rspec
+  ruby-2.4-rubocop:
+    docker:
+      - image: circleci/ruby:2.4
+    <<: *rubocop
+
+  # Ruby 2.5
+  ruby-2.5-rspec:
     docker:
       - image: circleci/ruby:2.5
-    steps:
-      - checkout
-      - run: bundle install
-      - run: rake
-  ruby-latest:
+    <<: *rspec
+  ruby-2.5-rubocop:
     docker:
-      - image: circleci/ruby:latest
-    steps:
-      - checkout
-      - run: bundle install
-      - run: rake
+      - image: circleci/ruby:2.5
+    <<: *rubocop
+
+  # JRuby
   jruby:
     docker:
       - image: circleci/jruby:9.1
     steps:
       - checkout
-      - run: bundle install
-      - run: rake
+      - run: bundle lock
+      - restore_cache:
+          keys:
+            - bundle-v1-{{ checksum "Gemfile.lock" }}
+            - bundle-v1-
+      - run: bundle install --path vendor/bundle
+      - save_cache:
+          key: bundle-v1-{{ checksum "Gemfile.lock" }}
+          paths:
+            - vendor/bundle
+      - restore_cache:
+          keys:
+            - rubocop-v1-{{ .Environment.CIRCLE_JOB }}
```

Comment:
Hmm, on second thought, this isn’t a good cache key. The Circle caches are immutable once written, and this cache key would only work if the cache could be appended to.